components/ui/Feed/FeedView.tsx

```diff
@@ -1,74 +1,48 @@
-import React, { useEffect, useRef, useState } from "react";
-import { ListingType, PostView, SortType } from "lemmy-js-client";
+import React, { useEffect, useMemo, useRef, useState } from "react";
+import { PostView } from "lemmy-js-client";
 import { HStack, useTheme, View } from "native-base";
 import { Button, RefreshControl, StyleSheet } from "react-native";
-import { useActionSheet } from "@expo/react-native-action-sheet";
-import { FlashList } from "@shopify/flash-list";
+import { FlashList, ListRenderItemInfo } from "@shopify/flash-list";
 import { useNavigation, useScrollToTop } from "@react-navigation/native";
-import { trigger } from "react-native-haptic-feedback";
 import { NativeStackNavigationProp } from "@react-navigation/native-stack";
-import {
-  IconBolt,
-  IconCalendar,
-  IconClockHour4,
-  IconDots,
-  IconFlame,
-  IconMessage,
-} from "tabler-icons-react-native";
 import FeedItem from "./FeedItem";
 import LoadingView from "../Loading/LoadingView";
 import FeedHeaderDropdownDrawer from "./FeedHeaderDropdownDrawer";
 import { useAppDispatch, useAppSelector } from "../../../store";
 import { selectFeed, setDropdownVisible } from "../../../slices/feed/feedSlice";
 import { UseFeed } from "../../hooks/feeds/useFeed";
-import LoadingFooter from "../Loading/LoadingFooter";
-import LoadingErrorFooter from "../Loading/LoadingErrorFooter";
-import { lemmyAuthToken, lemmyInstance } from "../../../lemmy/LemmyInstance";
 import LoadingErrorView from "../Loading/LoadingErrorView";
 import CompactFeedItem from "./CompactFeedItem/CompactFeedItem";
 import { selectSettings } from "../../../slices/settings/settingsSlice";
 import { ExtensionType, getLinkInfo } from "../../../helpers/LinkHelper";
-import HeaderIconButton from "../buttons/HeaderIconButton";
-import { IconCalendarWeek } from "../customIcons";
-import { showToast } from "../../../slices/toast/toastSlice";
 import NoResultView from "../common/NoResultView";
+import { Footer } from "./Footer";
+import { FeedSortButton } from "./FeedSortButton";
+import { Community, FeedOverflowButton } from "./FeedOverflowButton";
 
 interface FeedViewProps {
   feed: UseFeed;
   community?: boolean;
   header?: () => JSX.Element | null;
 }
 
-const SortIconType = {
-  TopDay: <IconCalendar />,
-  TopWeek: <IconCalendarWeek />,
-  Hot: <IconFlame />,
-  Active: <IconBolt />,
-  New: <IconClockHour4 />,
-  MostComments: <IconMessage />,
-};
-
 function FeedView({ feed, community = false, header }: FeedViewProps) {
   // State Props
   // TODO Handle this
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [endReached, setEndReached] = useState(false);
```

Comment:
I'm not familiar with the Lemmy API (honestly, I'd never heard of Lemmy until today), but ideally it would have some kind of pagination information. If that's the case, then we can derive from that instead of storing a `endReached` state.